vespa-feed-client/src/main/java/ai/vespa/feed/client/impl/JettyCluster.java

```diff
@@ -81,8 +82,11 @@ public void dispatch(HttpRequest req, CompletableFuture<HttpResponse> vessel) {
             Endpoint endpoint = findLeastBusyEndpoint(endpoints);
             try {
                 endpoint.inflight.incrementAndGet();
-                long reqTimeoutMillis = req.timeout() != null
-                        ? req.timeout().toMillis() * 11 / 10 + 1000 : IDLE_TIMEOUT.toMillis();
+                long reqTimeoutMillis = req.timeLeft().toMillis();
+                if (reqTimeoutMillis <= 0) {
+                    vessel.completeExceptionally(new TimeoutException("Operation timed out"));
```

Comment:
Consider adding timeout value to message (handy when user relies on default timeout value).